src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -24,8 +25,9 @@ internal sealed partial class JsonClassInfo
         // The properties on a POCO keyed on property name.
         public volatile Dictionary<string, JsonPropertyInfo> PropertyCache;
 
-        // Cache of properties by first JSON ordering. Use an array for highest performance.
-        private volatile PropertyRef[] _propertyRefsSorted = null;
+        // Cache of properties by first JSON ordering. Use an array (instead of List<T>) for highest performance.
+        private PropertyRef[] _propertyRefsSorted = null;
+        private object _cacheLock = new object();
```

Comment:
This should be readonly.

Reply:
Yep. Thanks

Reply:
The lock is no longer needed with the second commit.